util/progress/progressui/display.go

```diff
@@ -42,22 +43,33 @@ func WithDesc(text string, console string) DisplaySolveStatusOpt {
 	}
 }
 
-func DisplaySolveStatus(ctx context.Context, c console.Console, w io.Writer, ch chan *client.SolveStatus, opts ...DisplaySolveStatusOpt) ([]client.VertexWarning, error) {
-	modeConsole := c != nil
+// SolveStatusDisplay is a display that can show the updates to the solver status.
+type SolveStatusDisplay interface {
```

Comment:
Let's avoid public interfaces with private methods. I think all-public is fine here unless we completely want to hide it.

Reply:
The reason why I decided to make this have private methods is because the methods themselves aren't really useful or implementable for someone outside of this package. If we want to expose this to allow a custom `SolveStatusDisplay`, I think we need to do a larger refactor to expose more of this package. I'm a bit uncomfortable with doing that without a clear use case.

I think, if there's a compelling need to expose this more, we can defer what this interface should look like until then. Doing it this way allows us to implement the feature without making a promise to anyone using the library.

Reply:
I'm ok with `NewDisplay` just returning a struct and all this backend stuff being completely private.